src/mono/mono/mini/interp/transform.c

```diff
@@ -4189,17 +4224,25 @@ interp_emit_sfld_access (TransformData *td, MonoClassField *field, MonoClass *fi
 				td->last_ins->data [0] = get_data_item_index (td, field_class);
 		}
 	} else {
+		gboolean wide_data = is_data_item_wide_index (td);
```

Comment:
Logically there shouldn't be this side effect for other opcodes. I think it would make more sense to just get the required indexes with `get_data_item_index_wide` and if both indexes are short use the standard opcodes otherwise use the wide version.